pkg/sql/sqlbase/structured.go

```diff
@@ -3272,13 +3272,46 @@ func (desc *MutableTableDescriptor) performComputedColumnSwap(swap *ComputedColu
 	// oldCol still needs to have values written to it in case nodes read it from
 	// it with a TableDescriptor version from before the swap.
 	// To achieve this, we make oldCol a computed column of newCol.
-	oldColComputeExpr := tree.CastExpr{
-		Expr:       &tree.ColumnItem{ColumnName: tree.Name(oldCol.Name)},
-		Type:       oldCol.DatumType(),
-		SyntaxMode: tree.CastShort,
+	// If a USING EXPRESSION was provided, we cannot generally invert
+	// the EXPRESSION to insert into the old column and thus will force an
+	// error using the computed expression. Any inserts into the new column
+	// will fail until the old column is dropped.
+	if swap.InverseExpr != nil && *swap.InverseExpr == tree.Serialize(tree.DNull) {
```

Comment:
I meant that always serialize this erroring cast during planning and send it over in `InverseExpr`. If inverse expr != nil, then use it, otherwise create a basic cast here. Is that not possible?